tests/RobotFramework/tests/cumulocity/telemetry/child_device_telemetry.robot

```diff
@@ -254,8 +254,8 @@ Nested child devices support sending inventory data via tedge topic
     Execute Command    tedge mqtt pub --retain "te/device/${nested_child}///twin/type" ''
     Execute Command    tedge mqtt pub --retain "te/device/${nested_child}///twin/name" ''
     ${mo}=    Device Should Have Fragments    type
-    Should Be Equal    ${mo["type"]}    thin-edge.io-child
-    Should Be Equal    ${mo["name"]}    ${nested_child}
+    Should Be Equal    ${mo["type"]}    NewType
+    Should Be Equal    ${mo["name"]}    NewName
```

Comment:
I honestly don't know why we're repeating this same test for main device, child device, nested child device and with builtin bridge. The logic is the same for all cases. So, I'm even considering keeping these checks only for one case and removing the rest.

Reply:
I would not remove these tests as they are testing things which are different from the user point of view: twin data can be attached to the main device or any child devices. Running the same tests with and without the builtin doesn't harm.

The problem is that the test *code* is duplicated. Could the test be rewritten so the feature (setting twin data in the cloud) can be checked with a single test that is launched is several context (main vs child device, mosquitto vs builtin bridge)?

Reply:
Yeah, we could do something like that with test templates. Will create a follow-up task for that.

Reply:
Though sometimes abstracting the tests too much makes it more difficult to run selected tests (e.g. RobotFramework Templates don't actually allow you to run an individual test case using the VSCode plugin)...so sometimes copying/pasting is a more pragmatic approach (given the test cases don't change that much)

Reply:
Considering both the feedback, I went ahead with a far simpler approach in 45e1149 instead of using templates.